homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -0,0 +1,140 @@
+        ENERGY_KILO_WATT_HOUR)
+        }
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Required(CONF_CLOUD_ID): cv.string,
+    vol.Required(CONF_INSTALL_CODE): cv.string
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create the Eagle-200 sensor."""
+    ip_address = config[CONF_IP_ADDRESS]
+    cloud_id = config[CONF_CLOUD_ID]
+    install_code = config[CONF_INSTALL_CODE]
+
+    try:
+        eagle_reader = EagleReader(ip_address, cloud_id, install_code)
+    except (ConnectError, HTTPError, Timeout, ValueError) as error:
+        _LOGGER.error("Failed to connect during setup: %s", error)
+        return
+
+    eagle_data = EagleData(eagle_reader)
+    eagle_data.update()
+    monitored_conditions = list(SENSORS)
+    sensors = []
+    for condition in monitored_conditions:
+        sensors.append(EagleSensor(
+            eagle_data, condition, SENSORS[condition][0],
+            SENSORS[condition][1]))
+
+    add_entities(sensors)
+
+
+class EagleSensor(Entity):
+    """Implementation of the Rainforest Eagle-200 sensor."""
+
+    def __init__(
+            self, eagle_data, sensor_type, name, unit):
+        """Initialize the sensor."""
+        self.eagle_data = eagle_data
+        self._type = sensor_type
+        self._name = name
+        self._unit_of_measurement = unit
+        self._state = None
+
+    @property
+    def device_class(self):
+        """Return the power device class for the instantanous_demand sensor."""
+        if self._type == 'instantanous_demand':
+            return DEVICE_CLASS_POWER
+
+        return None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Get the energy information from the Rainforest Eagle."""
+        self.eagle_data.update()
+        data = self.eagle_data.data
+        self._state = EagleData.get_state(data, self._type)
+
+
+class EagleData:
+    """Get the latest data from the Eagle-200 device."""
+
+    def __init__(self, eagle_reader):
+        """Initialize the data object."""
+        self._eagle_reader = eagle_reader
+        self.data = {}
+
+    @Throttle(MIN_SCAN_INTERVAL)
+    def update(self):
+        """Get the latest data from the Eagle-200 device."""
+        try:
+            self.data = self._eagle_reader.update()
+            _LOGGER.debug("API data: %s", self.data)
+        except (ConnectError, HTTPError, Timeout, ValueError) as error:
+            _LOGGER.error("Unable to connect during update: %s", error)
+            self.data = None
+
+    @staticmethod
+    def get_state(data, sensor_type):
```

Comment:
Please remove the `data` parameter, and replace with `self`.

Reply:
Okay, thank you that is much cleaner!  Removed `data` and replaced with `self`.